### PR TITLE
Add stan package

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
This PR adds `stan` metricset. 

Still need to add a dashboard for `log` datastream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md#tips-for-building-integrations) and this pull request is aligned with them.
- [x] I have verified that all datasets collect metrics or logs.

## How to test this PR locally
Test the service manually using `elastic-package`:
1.  follow the instructions at https://github.com/elastic/elastic-package/blob/master/docs/howto/connect_service_with_elastic_stack.md#howto-connect-the-service-with-the-elastic-stack start a sample stan service. 
2. Install the stan integration from kibana. Use `/tmp/service_logs/stan.log*` for logs and `elastic-package-service_stan_1:8222` for metrics.
3. Verify that data are flowing for all datastreams and dashboards are working properly.

## Related issues

- Closes https://github.com/elastic/integrations/issues/363

## Screenshots
Screenshot 2021-01-15 at 12 29 04 PM
Screenshot 2021-01-15 at 12 29 33 PM
Screenshot 2021-01-18 at 3 32 46 PM
Screenshot 2021-01-15 at 3 02 40 PM
Screenshot 2021-01-15 at 1 38 35 PM
Screenshot 2021-01-18 at 3 28 19 PM

### CODE REVIEW

**Review by mtojek — COMMENTED** [review]
Could you please rebase it against the master branch? I pushed a new test runner for checking presence of package assets. It will verify if the package can be installed.

File: packages/stan/_dev/build/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# STAN integration
+
+This integration is used to collect logs and metrics from [STAN servers](https://github.com/nats-io/stan.go).
+The integration collects metrics from [STAN monitoring server APIs](https://github.com/nats-io/nats-streaming-server/blob/master/server/monitor.go).
+
+
+## Compatibility
+
+The STAN package is tested with Stan 0.15.1
```
Comment (mtojek) [inline packages/stan/_dev/build/docs/README.md:9]: nit: missing dot at the end

File: packages/stan/_dev/deploy/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+ARG STAN_VERSION=0.15.1
+FROM nats-streaming:$STAN_VERSION
+
+# build stage
+FROM golang:1.13-alpine3.11 AS build-env
+RUN apk --no-cache add build-base git mercurial gcc
+RUN cd src && go get -d github.com/nats-io/stan.go/
+RUN cd src/github.com/nats-io/stan.go/examples/stan-bench && git checkout tags/v0.5.2 && go build .
```
Comment (mtojek) [inline packages/stan/_dev/deploy/docker/Dockerfile:8]: nit: I wonder if there isn't a prebuilt image for this, but let's leave it if there isn't

File: packages/stan/_dev/deploy/docker/run.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+/nats-streaming-server -DV -l /var/log/stan/stan.log -m 8222 &
+sleep 2
+while true; do /stan-bench -np 0 -ns 100 -qgroup T -n 100000000 -ms 1024 foo; done
+#while true; do /stan-bench -np 10 -ns 10 -n 1000000000 -ms 1024 bar; done &
+
+# Make sure the container keeps running
+tail -f /dev/null
```
Comment (mtojek) [inline packages/stan/_dev/deploy/docker/run.sh:9]: is this line necessary considering the `while true` above?

File: packages/stan/kibana/visualization/stan-c3d1ab80-41a8-11e9-a4da-b1df688edbcd.json [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                        "show": true,
+                        "style": {},
+                        "title": {
+                            "text": "Sum of Message Bytes"
+                        },
+                        "type": "value"
+                    }
+                ]
+            },
+            "title": " Bytes Timeline [Logs STAN]",
```
Comment (mtojek) [inline packages/stan/kibana/visualization/stan-c3d1ab80-41a8-11e9-a4da-b1df688edbcd.json:104]: nit: unnecessary space

File: packages/stan/kibana/visualization/stan-c3d1ab80-41a8-11e9-a4da-b1df688edbcd.json [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            "searchSourceJSON": {
+                "filter": [],
+                "indexRefName": "kibanaSavedObjectMeta.searchSourceJSON.index",
+                "query": {
+                    "language": "kuery",
+                    "query": "data_stream.dataset: stan.log"
+                }
+            }
+        },
+        "title": " Bytes Timeline [Logs STAN]",
```
Comment (mtojek) [inline packages/stan/kibana/visualization/stan-c3d1ab80-41a8-11e9-a4da-b1df688edbcd.json:14]: nit: actually this pattern is repeated in few places:

```
"title": "<space> 
```

File: packages/stan/manifest.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    type: image/svg+xml
+format_version: 1.0.0
+license: basic
+categories:
+  - message_queue
+conditions:
+  kibana.version: ">=7.11.0"
+screenshots:
+  - src: /img/metricbeat-stan-overview.png
+    title: Metricbeat STAN Dashboard
```
Comment (mtojek) [inline packages/stan/manifest.yml:20]: Metrics

File: packages/stan/manifest.yml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    size: 1829x447
+    type: image/png
+  - src: /img/filebeat-stan-overview.png
+    title: Filebeat STAN Dashboard
```
Comment (mtojek) [inline packages/stan/manifest.yml:24]: Logs, not filebeat

**Review by mtojek — COMMENTED** [review]
Great, all tests are passing including assets: https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations/detail/PR-532/13/tests

I left some comments.

File: packages/stan/manifest.yml [hunk 8]
```diff
@@ -17,11 +17,11 @@ conditions:
   kibana.version: ">=7.11.0"
 screenshots:
   - src: /img/metricbeat-stan-overview.png
```
Comment (mtojek) [inline packages/stan/manifest.yml:19]: nit: I think you can fix filenames too.

**Review by mtojek — APPROVED** [review]
Please wait for green. If it passes, fee free to merge this PR. LGTM!

File: packages/stan/_dev/deploy/docker/Dockerfile (hunk 2 above)
Comment (ChrsMark) [reply packages/stan/_dev/deploy/docker/Dockerfile:8]: Didn't find anything with a quick search maintained by NATS project and I would avoid using a random third party custom image (if any) so better to handle it on our own. In the future these images will be cached, no?

Comment (mtojek) [reply packages/stan/_dev/deploy/docker/Dockerfile:8]: SGTM! We'll cache image when it becomes the problem. So far image building doesn't take much time.

**Pull request merged**